core/CMakeLists.txt

```diff
@@ -93,7 +93,7 @@ endif()
 
 # Link with thread library, unless we are on the Zephyr platform.
 if(NOT DEFINED LF_SINGLE_THREADED OR DEFINED LF_TRACE)
-    if (NOT PLATFORM_ZEPHYR)
+    if (NOT (PLATFORM_ZEPHYR OR ${CMAKE_SYSTEM_NAME} STREQUAL "Rp2040"))
```

Comment:
I tried `if (NOT (PLATFORM_ZEPHYR OR PLATFORM_RP2040))` but it didn't work. Not sure how `PLATFORM_ZEPHYR` works here.

Reply:
todo: fix, same as https://github.com/lf-lang/reactor-c/blob/e8a0b97917c8ef146cd074e67aa69df43f344bb6/core/platform/CMakeLists.txt#L24C1-L24C30